### PR TITLE
SUBMARINE-323. Add how to release guide and scripts.

### PR DESCRIPTION
**Pull request by tangzhankun**
### What is this PR for?
Submarine release scripts and guide

### What type of PR is it?
Feature

### Todos
* [1] - Automate steps in cicd images

### What is the Jira issue?
https://issues.apache.org/jira/browse/SUBMARINE-323

### How should this be tested?
When a committer is preparing a new release, this can be verified.
Zhankun has gone through the guide with the scripts. No big issue found yet.

### Questions:
* Does the licenses files need update? No
* Is there breaking changes for older versions? No
* Does this needs documentation? Yes

### CODE REVIEW

File: dev-support/cicd/create_release.sh [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+RELEASE_VERSION="$1"
+GIT_TAG="$2"
+
+function compile_src_and_bin() {
+  cd ${WORKING_DIR}/submarine
+  echo "git submodule update --init --recursive"
+  git submodule update --init --recursive
+  echo "git submodule update --recursive"
+  git submodule update --recursive
```
Comment (liuxunorg) [inline dev-support/cicd/create_release.sh]: submodule tony will be delete. https://github.com/apache/submarine/pull/173

File: dev-support/cicd/publish_release.sh [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+    echo "ERROR: Snapshots must have a version containing 'SNAPSHOT'"
+    echo "ERROR: You gave version '$RELEASE_VERSION'"
+    exit 1
+  fi
+
+  tmp_repo="$(mktemp -d /tmp/submarine-repo-XXXXX)"
+  echo "git submodule update --init --recursive"
+  git submodule update --init --recursive
+  echo "git submodule update --recursive"
+  git submodule update --recursive
```
Comment (liuxunorg) [inline dev-support/cicd/publish_release.sh]: submodule tony will be delete. https://github.com/apache/submarine/pull/173

File: dev-support/cicd/publish_release.sh [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+  mvn versions:set -DnewVersion=$RELEASE_VERSION
+  tmp_settings="tmp-settings.xml"
+  echo "<settings><servers><server>" > $tmp_settings
+  echo "<id>apache.snapshots.https</id><username>$ASF_USERID</username>" >> $tmp_settings
+  echo "<password>$ASF_PASSWORD</password>" >> $tmp_settings
+  echo "</server></servers></settings>" >> $tmp_settings
+
+  #Disable TonY mvn deploy
+  #tony_pom="./submodules/tony/pom.xml"
+  mvn --settings $tmp_settings -Dmaven.repo.local="${tmp_repo}" -DskipTests deploy
```
Comment (liuxunorg) [inline dev-support/cicd/publish_release.sh]: submodule tony will be delete. https://github.com/apache/submarine/pull/173

File: dev-support/cicd/create_release.sh (hunk 1 above)
Comment (tangzhankun) [reply dev-support/cicd/create_release.sh]: Thanks for reminding this. Fixed.

**Comment by tangzhankun** [comment]
@liuxunorg, @yuanzac , @jiwq the Travis error seems unrelated. Could you please help to review it?

**Comment by liuxunorg** [comment]
Will merge if no more comments.

**Pull request closed without merging**